install/installer/pkg/components/server/deployment.go

```diff
@@ -330,19 +330,21 @@ func deployment(ctx *common.RenderContext) ([]runtime.Object, error) {
 	}
 
 	// admin secret
-	volumes = append(volumes, corev1.Volume{
-		Name: "admin-login-key",
-		VolumeSource: corev1.VolumeSource{
-			Secret: &corev1.SecretVolumeSource{
-				SecretName: AdminSecretName,
+	if ctx.Config.AdminLoginSecret != nil {
+		volumes = append(volumes, corev1.Volume{
+			Name: "admin-login-key",
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName: ctx.Config.AdminLoginSecret.Name,
+				},
 			},
-		},
-	})
-	volumeMounts = append(volumeMounts, corev1.VolumeMount{
-		Name:      "admin-login-key",
-		MountPath: AdminSecretMountPath,
-		ReadOnly:  true,
-	})
+		})
+		volumeMounts = append(volumeMounts, corev1.VolumeMount{
+			Name:      "admin-login-key",
+			MountPath: AdminSecretMountPath,
+			ReadOnly:  true,
+		})
+	}
```

Comment:
Is that key ever optional?

May make it easier if use the `Optional: true` property on the volume-mount, that way you can avoid the extra conditional on the config render

Reply:
Thx for sharing! TIL about that Optional flag. But so far I like the failure mode better (server pod not coming up, vs single error log line). If we want to change that later, we can always revise. :lotus_position: